### PR TITLE
:seedling: logicalcluster_deletion_controller uses committer

### PR DESCRIPTION
**Pull request by csams**
## Summary
Update the logicalcluster_deletion_controller to use the committer package.

Signed-off-by: Christopher Sams <csams@redhat.com>

## Related issue(s)
#2635

### CODE REVIEW

**Review by ncdc — APPROVED** [review]

File: pkg/reconciler/core/logicalclusterdeletion/logicalcluster_deletion_controller.go [hunk 1]
```diff
@@ -236,49 +245,15 @@ func (c *Controller) process(ctx context.Context, key string) error {
...
-	if err != nil {
-		return fmt.Errorf("failed to create patch for logical cluster %s: %w", new.Name, err)
+	oldResource := &Resource{ObjectMeta: logicalCluster.ObjectMeta, Spec: &logicalCluster.Spec, Status: &logicalCluster.Status}
+	newResource := &Resource{ObjectMeta: logicalClusterCopy.ObjectMeta, Spec: &logicalClusterCopy.Spec, Status: &logicalClusterCopy.Status}
+	if err := c.commit(ctx, oldResource, newResource); err != nil {
+		errs = append(errs, err)
 	}
 
-	logger.V(2).Info("patching LogicalCluster", "patch", string(patchBytes))
-	_, err = c.kcpClusterClient.Cluster(logicalcluster.From(new).Path()).CoreV1alpha1().LogicalClusters().Patch(ctx, new.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{}, "status")
```
Comment (csams) [inline pkg/reconciler/core/logicalclusterdeletion/logicalcluster_deletion_controller.go:280]: @ncdc the logicalcluster used to scope the patch here is gotten from the `new` resource whereas in the committer it uses the old resource: https://github.com/kcp-dev/kcp/blob/main/pkg/reconciler/committer/committer.go#L63-L64. Should committer use the new resource instead?

**Review by csams — COMMENTED** [review]
Question about logical cluster usage in the old code vs. the committer.

**Comment by ncdc** [comment]
Flake #2479

**Comment by ncdc** [comment]
Flake #2479

**Comment by ncdc** [comment]
#2479 again

**Pull request merged**